xbmc/interfaces/builtins/LibraryBuiltins.cpp

```diff
@@ -160,18 +162,89 @@ static int ExportLibrary(const std::vector<std::string>& params)
     }
     else
     {
-      if (URIUtils::HasSlashAtEnd(path))
-        path = URIUtils::AddFileToFolder(path, "musicdb.xml");
-      CMusicDatabase musicdatabase;
-      musicdatabase.Open();
-      musicdatabase.ExportToXML(path, singleFile, thumbs, overwrite);
-      musicdatabase.Close();
+      CLibExportSettings settings;
+      // ELIBEXPORT_SINGLEFILE, ELIBEXPORT_ALBUMS + ELIBEXPORT_ALBUMARTISTS by default
+      settings.m_strPath = path;
+      if (!singleFile)
+        settings.m_exporttype = ELIBEXPORT_TOLIBRARYFOLDER;
+      settings.m_artwork = thumbs;
+      settings.m_overwrite = overwrite;
+      // Export music library (not showing progress dialog)
+      CMusicLibraryQueue::GetInstance().ExportLibrary(settings, false);
     }
   }
 
   return 0;
 }
 
+/*! \brief Export a library with extended parameters
+Avoiding breaking change to original ExportLibrary routine parameters
+*  \param params The parameters.
+*  \details params[0] = "video" or "music".
+*           params[1] = export type "singlefile", "separate", or "library".
+*           params[2] = path of destination folder.
+*           params[3,...] = "unscraped" to include unscraped items
+*           params[3,...] = "overwrite" to overwrite exitsing files.
+*           params[3,...] = "artwork" to include images such as thumbs and fanart.
+*           params[3,...] = "skipnfo" to not include nfo files (just art).
+*           params[3,...] = "ablums" to include albums.
+*           params[3,...] = "albumartists" to include album artists.
+*           params[3,...] = "songartists" to include song artists.
+*           params[3,...] = "otherartists" to include other artists.
+*/
+static int ExportLibrary2(const std::vector<std::string>& params)
```

Comment:
(non-void) function misses a return at the end

Reply:
thanks for fixing
As this function only returns `-1` and `0` you could switch from int return type to bool.

Reply:
@DaveTBlake have you overlooked the above comment, with maybe switching return type to bool?

Reply:
Not overlooked. I have a bigger question over if I can make a breaking change to the builtin interface for `ExportLibrary()` rather than have `ExportLibrary2()`at all, so this routine would go. Would change return type then potentially

Reply:
Nothing crucial depends on the builtin. Just change imo we never promised a stable api. The return tyoe is however not easily changeable due to the function ptr table usage.